### PR TITLE
BFD-3603: Change careTeam[].qualification to use taxonomy

### PR DESCRIPTION
**Pull request by dondevun**
**JIRA Ticket:**
BFD-3603

### What Does This PR Do?

This PR changes careTeam[].qualification to use taxonomy, instead of prvdr_spclty. The npiOrgLookup module has been refactored to include Practitioner NPIs, instead of just Organization NPIs, and the taxonomy mappings have been downloaded and put into the resources folder to use as a source during lookup.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

### What Security Implications Does This PR Have?

Please indicate if this PR does any of the following:  

* Adds any new software dependencies
* Modifies any security controls
* Adds new transmission or storage of data
* Any other changes that could possibly affect security? 

* [x] I have considered the above security implications as it relates to this PR. (If one or more of the above apply, it cannot be merged without the ISSO or team security engineer's (`@sb-benohe`) approval.) 

### Validation

Tested against golden bene for all claim types.

### CODE REVIEW

File: apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java [hunk 1]
```diff
@@ -262,6 +266,36 @@ private static void buildNPIResource(
...
+   * @throws IOException exception thrown.
+   */
+  public static Map<String, String> processTaxonomyDescriptions() throws IOException {
+    ClassLoader classLoader = DataUtilityCommons.class.getClassLoader();
+    File taxonomyFile =
+        new File(
+            Objects.requireNonNull(classLoader.getResource("nucc_taxonomy_240.csv")).getFile());
+    Map<String, String> taxonomyMapping = new HashMap<>();
+    try (FileInputStream is = new FileInputStream(taxonomyFile);
+        BufferedReader reader = new BufferedReader(new InputStreamReader(is))) {
```
Comment (aschey-forpeople) [inline apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java:282]: Would it be easier to use a CSV reader here?

Comment (dondevun) [reply apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java:282]: Maybe, but if we did that, then we should probably refactor the rest of the code in this module, as well, to use the CSV reader -- this follows the same pattern that already exists to read the NPI file. I'm not sure if introducing a library and refactoring all of this code is the best course of action, but I'm open to a conversation about it.

**Review by MahiFentaye — DISMISSED** [review]

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/NPIOrgLookupIT.java [hunk 2]
```diff
@@ -15,7 +15,7 @@ public class NPIOrgLookupIT {
   public void VerifyAValidNPIOrg() throws IOException {
     NPIOrgLookup npiOrgLookup = NPIOrgLookup.createNpiOrgLookup();
     Optional<String> orgDisplay = npiOrgLookup.retrieveNPIOrgDisplay(Optional.of("1497758544"));
-    assertEquals("CUMBERLAND COUNTY HOSPITAL SYSTEM", orgDisplay.get());
+    assertEquals("CUMBERLAND COUNTY HOSPITAL SYSTEM, INC", orgDisplay.get());
```
Comment (dondevun) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/NPIOrgLookupIT.java:18]: Adding the CSV reader uncovered a bug where org names with commas were not displaying correctly.

File: apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java (hunk 1 above)
Comment (dondevun) [reply apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java:295]: Good call on the CSV reader. CSV parsing bugs were uncovered.

Comment (aschey-forpeople) [reply apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java:295]: Awesome, thanks for looking at that.

File: apps/bfd-server/bfd-server-war/src/test/resources/endpoint-responses/v2/eobByPatientIdAll.json [hunk 3]
```diff
@@ -1050,13 +1050,6 @@
             "code" : "performing",
             "display" : "Performing provider"
           } ]
-        },
-        "qualification" : {
```
Comment (aschey-forpeople) [inline apps/bfd-server/bfd-server-war/src/test/resources/endpoint-responses/v2/eobByPatientIdAll.json:1054]: Are all of these supposed to be removed now? Might be nice to update these tests to include the new fields.

**Comment by dondevun** [comment]
The new NPI enrichment story will refactor a lot of this code, so I think I'm going to abandon this, and continue the work in the new NPI enrichment story.

File: apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java [hunk 4]
```diff
@@ -271,45 +315,42 @@ private static void buildNPIResource(
...
+        String taxonomyCode = csvRecord.get(TAXONOMY_CODE_FIELD);
         // entity type code 2 is organization
-        if (!Strings.isNullOrEmpty(entityTypeCode) && Integer.parseInt(entityTypeCode) == 2) {
-          out.write(npi + "\t" + orgName);
-          out.newLine();
+        if (!Strings.isNullOrEmpty(entityTypeCode)) {
+          if (Integer.parseInt(entityTypeCode) == 2 && !Strings.isNullOrEmpty(orgName)) {
+            out.write(npi + "\t" + orgName);
+            out.newLine();
+          } else if (!Strings.isNullOrEmpty(taxonomyCode)) {
```
Comment (dondevun) [inline apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java:350]: This should resolve the empty rows in the constructed file.

**Review by MahiFentaye — DISMISSED** [review]

File: apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java (hunk 4 above)
Comment (aschey-forpeople) [reply apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java:350]: What's the reason that some of the org names are empty?

Comment (aschey-forpeople) [reply apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java:350]: Makes sense, thanks!

File: apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java (hunk 1 above)
Comment (aschey-forpeople) [reply apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java:295]: Actually, I don't think this is always the case. I added this line
```java
 if (Integer.parseInt(entityTypeCode) == 2 && Strings.isNullOrEmpty(orgName)) {
            System.out.println("empty org");
 }
```
and it got a few hits before I stopped it.

File: apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java (hunk 4 above)
Comment (aschey-forpeople) [reply apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java:350]: Actually, I don't think this is always the case. I added this line
```java
 if (Integer.parseInt(entityTypeCode) == 2 && Strings.isNullOrEmpty(orgName)) {
            System.out.println("empty org");
 }
```
and it got a few hits before I stopped it.

**Review by MahiFentaye — APPROVED** [review]

**Pull request merged**